osidb/serializer.py

```diff
@@ -402,6 +404,79 @@ class Meta:
         fields = ["alerts"]
 
 
+class AuditSerializer(serializers.ModelSerializer):
+
+    pgh_data = serializers.SerializerMethodField()
+
+    def get_pgh_data(self, obj):
+        # remove acls from entity snapshot
+        data = obj.pgh_data
+        data.pop("acl_read")
+        data.pop("acl_write")
+        return obj.pgh_data
+
+    class Meta:
+        model = Events
+        fields = [
+            "pgh_created_at",
+            "pgh_slug",
+            "pgh_obj_model",
+            "pgh_obj_id",
+            "pgh_label",
+            "pgh_context",
+            "pgh_diff",
+            "pgh_data",
+        ]
+
+
+class HistoricalEventSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Events
+        fields = [
+            "pgh_created_at",
+            "pgh_slug",
+            "pgh_label",
+            "pgh_context",
+            "pgh_diff",
+        ]
+
+
+class HistoryMixinSerializer(serializers.ModelSerializer):
+
+    history = serializers.SerializerMethodField()
+
+    def __init__(self, *args, **kwargs):
+        # Instantiate the superclass normally
+        super().__init__(*args, **kwargs)
+
+        request = self.context.get("request")
+        include_history_param = None
+        # Get include history from request
+        if request:
+            include_history_param = request.query_params.get("include_history")
```

Comment:
Suggestion: You might want to describe this query parameter in OpenAPI schema the same way as we did with the `include_fields` or `exclude_fields`, see api_views.py for that.